pkg/dataobj/internal/dataset/reader_basic.go

```diff
@@ -224,7 +224,10 @@ func (pr *basicReader) fill(ctx context.Context, columns []Column, s []Row) (n i
 
 			columnRead := columnRow - startRow
 			for i := columnRead; i < int64(maxRead); i++ {
-				s[n+int(i)].Values[columnIndex] = Value{}
+				// Reset values to 0 without discarding any memory they are pointing to
+				if !s[n+int(i)].Values[columnIndex].IsNil() {
+					s[n+int(i)].Values[columnIndex].num = 0
+				}
```

Comment:
nit: I'd rather the internal structure of Value not be used by other parts of the code, primarily to keep dependencies between code to exposed APIs; can we have some kind of `Zero()` or `Reset()` method on Value instead?